Patch by MSP-Greg
Allows using runner.rb from install
diff --git a/tool/test/runner.rb b/tool/test/runner.rb
index 20ae9aa1a3..b13a0c976a 100644
--- a/tool/test/runner.rb
+++ b/tool/test/runner.rb
@@ -17,12 +17,19 @@
 $LOAD_PATH << src_testdir
 $LOAD_PATH.unshift "#{tool_dir}/lib"
 
-# Get bundled gems on load path
-Dir.glob("#{src_testdir}/../gems/*/*.gemspec")
-  .reject {|f| f =~ /minitest|test-unit|power_assert/ }
-  .map {|f| $LOAD_PATH.unshift File.join(File.dirname(f), "lib") }
+# Get bundled gems on load path, check whether running from install
+abi = RbConfig::CONFIG['ruby_version']
+if Dir.exist? (gem_dir = "#{RbConfig::TOPDIR}/lib/ruby/gems/#{abi}/gems")
+  Dir.glob("#{gem_dir}/*")
+    .reject { |f| f[/\/(minitest|test-unit|power_assert)-/] || !Dir.exist?("#{f}/lib") }
+    .each   { |f| $LOAD_PATH.unshift "#{f}/lib" }
+else
+  Dir.glob("#{src_testdir}/../gems/*/*.gemspec")
+    .reject {|f| f =~ /minitest|test-unit|power_assert/ }
+    .each {|f| $LOAD_PATH.unshift File.join(File.dirname(f), "lib") }
+end
 
-require 'test/unit'
+require_relative '../lib/test/unit'
 
 module Gem
 end
